File: scripts/datasim.py
#! /usr/bin/python


# https://pypi.python.org/pypi/paho-mqtt/1.1
import paho.mqtt.client as mqtt
import time
from random import randint
import json

def on_connect(client, userdata, flags, rc):
    print('connected')

def on_publish(client, userdata, mid):
    print('published')

def generate_telemetry():
    telemetry = {}
    telemetry['altitude'] = randint(0, 100)
    telemetry['power'] = randint(0, 100)
    telemetry['temperature'] = randint(0, 100)
    telemetry['latitude'] = randint(27000000, 29000000) / 1000000.0
    telemetry['longitude'] = randint(-82000000, -80000000) / 1000000.0
    telemetry['eventCode'] = randint(0, 15)
    return telemetry


client = mqtt.Client()
client.on_connect = on_connect
client.on_publish = on_publish
#client.on_message = on_message
client.connect('localhost', 1883, 60)
#client.loop_start()
while True:
    # generate random data
    try:
        time.sleep(0.5)
        client.loop(0.01)
        tjson = json.dumps(generate_telemetry())
        print(tjson)
        client.publish('telemetry', payload=tjson, qos=0, retain=True)
    except KeyboardInterrupt:
        #client.loop_stop()
        print('disconnecting...')
        break
client.disconnect()

File: scripts/datasim.js
const mqtt = require('mqtt');

var client = mqtt.connect('ws://localhost:3000');

setInterval(function() {
    var t = generateTelemetry();
    client.publish('telemetry', JSON.stringify(t), {
        retain: true
    });
}, 1000);




function generateTelemetry() {
    return {
        latitude: randInt(28550000, 28500000) / 1000000.0,
        longitude: randInt(-81250000, -81200000) / 1000000.0,
        altitude: randInt(0, 10),
        power: randInt(60, 75),
        temperature: randInt(70, 80),
        heading: randInt(0, 360),
        eventCode: randInt(0, 15)
    }
}

function randInt(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min + 1)) + min;
}